src/main/java/io/getstream/chat/java/models/Permission.java

```diff
@@ -51,77 +54,22 @@
     @NotNull private String expression;
   }
 
-  public enum Resource {
-    @JsonProperty("CreateChannel")
-    CREATE_CHANNEL,
-    @JsonProperty("CreateDistinctChannelForOthers")
-    CREATE_DISTINCT_CHANNEL_FOR_OTHERS,
-    @JsonProperty("ReadChannel")
-    READ_CHANNEL,
-    @JsonProperty("UpdateChannelMembers")
-    UPDATE_CHANNEL_MEMBERS,
-    @JsonProperty("RemoveOwnChannelMembership")
-    REMOVE_OWN_CHANNEL_MEMBERSHIP,
-    @JsonProperty("UpdateChannel")
-    UPDATE_CHANNEL,
-    @JsonProperty("UseFrozenChannel")
-    USE_FROZEN_CHANNEL,
-    @JsonProperty("UpdateUserRole")
-    UPDATE_USER_ROLE,
-    @JsonProperty("DeleteChannel")
-    DELETE_CHANNEL,
-    @JsonProperty("CreateMessage")
-    CREATE_MESSAGE,
-    @JsonProperty("UpdateMessage")
-    UPDATE_MESSAGE,
-    @JsonProperty("PinMessage")
-    PIN_MESSAGE,
-    @JsonProperty("DeleteMessage")
-    DELETE_MESSAGE,
-    @JsonProperty("RunMessageAction")
-    RUN_MESSAGE_ACTION,
-    @JsonProperty("MuteUser")
-    MUTE_USER,
-    @JsonProperty("BanUser")
-    BAN_USER,
-    @JsonProperty("UploadAttachment")
-    UPLOAD_ATTACHMENT,
-    @JsonProperty("DeleteAttachment")
-    DELETE_ATTACHMENT,
-    @JsonProperty("AddLinks")
-    ADD_LINKS,
-    @JsonProperty("CreateReaction")
-    CREATE_REACTION,
-    @JsonProperty("DeleteReaction")
-    DELETE_REACTION,
-    @JsonProperty("SendCustomEvent")
-    SEND_CUSTOM_EVENT,
-    @JsonProperty("SkipMessageModeration")
-    SKIP_MESSAGE_MODERATION,
-    @JsonProperty("UseCommands")
-    USE_COMMANDS,
-    @JsonProperty("EditUser")
-    EDIT_USER,
-    @JsonProperty("ReadMessageFlags")
-    READ_MESSAGE_FLAGS,
-    @JsonProperty("*")
-    ALL,
-    @JsonEnumDefaultValue
-    UNKNOWN
-  }
-
   @Builder(
       builderClassName = "PermissionCreateRequest",
       builderMethodName = "",
       buildMethodName = "internalBuild")
   public static class PermissionCreateRequestData {
+    @NotNull
+    @JsonProperty("id")
+    private String id;
+
     @Nullable
```

Comment:
This should probably be `@NotNull` also

Reply:
That's correct. Could you push it too?

Reply:
@ferhatelmas Pushed! Sorry for delay, I was kind of expecting some comments and was going to do it all in one pass 👍